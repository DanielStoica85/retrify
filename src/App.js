import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import { Provider } from 'react-redux';

import Navigation from './components/layout/Navigation';
import Dashboard from './components/dashboard/Dashboard';
import Login from './components/auth/Login';
import Register from './components/auth/Register';
import RetroDetails from './components/retros/RetroDetails';
import AddRetro from './components/retros/AddRetro';

import configureStore from './store/configureStore';

import { addRetro } from './actions/retros';

const store = configureStore();
console.log(store.getState());

store.dispatch(addRetro({ title: 'First', author: 'Daniel' }));
store.dispatch(addRetro({ title: 'Second', author: 'Gica' }));

function App() {
    return (
        <Provider store={store}>
            <BrowserRouter>
                <div className="App">
                    <Navigation />
                    <Switch>
                        <Route exact path="/" component={Dashboard} />
                        <Route path="/login" component={Login} />
                        <Route path="/register" component={Register} />
                        <Route path="/retro/:id" component={RetroDetails} />
                        <Route path="/retro" component={AddRetro} />
                    </Switch>
                </div>
            </BrowserRouter>
        </Provider>
    );
}

export default App;
