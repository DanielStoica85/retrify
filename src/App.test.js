import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

describe('App component', () => {
    it('matches the snapshot.', () => {
        const component = renderer.create(<App />).toJSON();
        expect(component).toMatchSnapshot();
    });
});
