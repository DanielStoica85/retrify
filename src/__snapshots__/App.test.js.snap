// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot. 1`] = `
<div
  className="App"
>
  <div>
    <nav
      className="mb-5 navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          Retrify
        </a>
        <button
          className="navbar-toggler"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          style={Object {}}
        >
          <ul
            className="ml-auto navbar-nav"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/register"
                onClick={[Function]}
              >
                Sign Up
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/retro"
                onClick={[Function]}
              >
                New Retro
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Log Out
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                <span
                  className="badge badge-danger badge-pill"
                >
                  D.S.
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="dashboard container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-12 col-md-6"
      >
        <div
          className="retros-list section"
        >
          <div>
            <div
              className="mt-3 card"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Some title
                </div>
                <div
                  className="card-subtitle"
                >
                  Added by 
                  Some author
                </div>
              </div>
              <div
                className="card-footer"
                style={
                  Object {
                    "backgroundColor": "lightgray",
                    "borderColor": "lightgray",
                  }
                }
              >
                May 10th, 2019
              </div>
            </div>
          </div>
          <div>
            <div
              className="mt-3 card"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Some title
                </div>
                <div
                  className="card-subtitle"
                >
                  Added by 
                  Some author
                </div>
              </div>
              <div
                className="card-footer"
                style={
                  Object {
                    "backgroundColor": "lightgray",
                    "borderColor": "lightgray",
                  }
                }
              >
                May 10th, 2019
              </div>
            </div>
          </div>
          <div>
            <div
              className="mt-3 card"
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title"
                >
                  Some title
                </div>
                <div
                  className="card-subtitle"
                >
                  Added by 
                  Some author
                </div>
              </div>
              <div
                className="card-footer"
                style={
                  Object {
                    "backgroundColor": "lightgray",
                    "borderColor": "lightgray",
                  }
                }
              >
                May 10th, 2019
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-sm-12 col-md-5 offset-md-1"
      >
        <div>
          <p>
            Notifications
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
